lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java

```diff
@@ -77,13 +83,23 @@ public static String defaultVersion() {
 
 		/** Are the files being linted Kotlin script files. */
 		private final boolean isScript;
+		private final String pkg;
 		/** The jar that contains the eclipse formatter. */
 		final JarState jarState;
 		private final TreeMap<String, String> userData;
 
 		State(String version, Provisioner provisioner, boolean isScript, Map<String, String> userData) throws IOException {
 			this.userData = new TreeMap<>(userData);
-			this.jarState = JarState.from(MAVEN_COORDINATE + version, provisioner);
+			String coordinate;
+			Matcher matcher = VERSION_PRE_0_32.matcher(version);
+			if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 32) {
+				coordinate = MAVEN_COORDINATE_PRE_0_32;
+				this.pkg = PACKAGE_PRE_0_32;
+			} else {
+				coordinate = MAVEN_COORDINATE;
+				this.pkg = PACKAGE;
+			}
```

Comment:
What happens when a SNAPSHOT is passed? If it doesn't match the regex, I suggest ensuring that this fails to match forward towards the newer version instead of failing back towards an older configuration

Reply:
It defaults to the newer configuration if the regex is not matched. And the regex doesn't include start- and end-of-string delimiters, so it should still match the regex if SNAPSHOT is at the end of the version.

Reply:
I will add a test for SNAPSHOT versions.

Reply:
It might be hard to make a SNAPSHOT test deterministic, I'm satisfied by the current "fail to newer" behavior.  I just restarted CI to see if the previous failure was intermittent or not.